libdevcore/Log.cpp

```diff
@@ -60,7 +60,7 @@ LogOutputStreamBase::LogOutputStreamBase(char const* _id, std::type_info const*
 	m_autospacing(_autospacing),
 	m_verbosity(_v)
 {
-	Guard l(x_logOverride);
+	x_logOverride.lock();
```

Comment:
I think easier fix would be to change
```cpp
void dev::simpleDebugOut(std::string const& _s)
{
     std::cerr << _s << '\n';
}
```
to
```cpp
void dev::simpleDebugOut(std::string _s)
{
    _s.push_back('\n')
    std::cerr << _s;
}
```

We have the issue now, because the endline char is sent to output stream separately.

Reply:
You may be right, and that was my first approach. But then I thought, there is no guarantee that everything in the formatting part is and will remain re-entrant

Reply:
Ok I'll close and push a clean update